cpp/oneapi/dal/detail/mpi/communicator.hpp

```diff
@@ -146,6 +149,53 @@ class mpi_communicator_impl : public via_host_interface_selector<MemoryAccessKin
         return default_root_;
     }
 
+    bool get_mpi_gpu_support() override {
+        // TODO: explore additional conditions (i.e. GPU type, data parallel, etc.)
+        // TODO: make this function bool convert() and instead determine if conversion necessary?
+
+        // Check libmpi.so for symbol
+        void* handle = dlopen("libmpi.so", RTLD_LAZY);
```

Comment:
may be it makes sense to ask MPI team about the adding public method for getting this info through a mpi function? Due to its little bit hardcoded and could be not flexible if MPI change smth

Reply:
This was their recommended approach, although I agree that it may be worth seeing if they can add something as easy to use as with CCL

Reply:
@Vika-F Hi! Can you please review this part of the code?

Reply:
Hi @irozanova, thanks for the answer! In terms of backward compatibility, I guess that we can add versioning and for previous versions don't use GPU offloading. The question is, is it possible to somehow get the information about "MPIX_Query_ze_support" via some of MPI function?

Reply:
> The question is, is it possible to somehow get the information about "MPIX_Query_ze_support" via some of MPI function?

No, there are no any functions, which provide information about other functions.

Reply:
This approach is also compatible with MPICH and other MPI backends, as they have the same symbol to indicate if offloading is supported

Reply:
Taking into account the fact that `MPIX_Query_ze_support` is only supported in Intel MPI and only starting from 2021.12, I think it would be better to leave the code as is.

Reply:
MPIX_Query_ze_support is supported also by MPICH implementation
https://www.mpich.org/static/docs/latest/www3/MPIX_Query_ze_support.html